src/pub.c

```diff
@@ -104,7 +104,7 @@ natsConn_publish(natsConnection *nc, natsMsg *msg, const char *reply, bool direc
     // accessing the headers. It should still be considered having headers.
     if ((msg->headers != NULL) || natsMsg_needsLift(msg))
     {
-        if (!nc->info.headers)
```

Comment:
nit: 0/5 update the comment with the comment from this PR to explain why `initc` is excluded?